### PR TITLE
Restrictions

### PR DESCRIPTION
**Pull request by regineheberlein**

### CODE REVIEW

File: data_fixes/change_restriction/fix_restriction.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    #change restriction note here. Takes a string.
+    #accessrestrict[0]['subnotes'][0]['content'] = row['restriction_note']
+    #change the end date here
+    #accessrestrict[0]['rights_restriction']['end'] = row['end_date'] unless row['end_date'].empty?
+    post = @client.post(row['uri'], record.to_json)
+    #write to log
+    File.write(log, post.body, mode: 'a')
+    puts post.body
+    rescue Exception => msg
+    error = "Process ended: #{Time.now} with error '#{msg.class}: #{msg.message}''"
```
Comment (carolyncole) [inline data_fixes/change_restriction/fix_restriction.rb:29]: You are not doing anything with the error.  Maybe just a puts, but I would think you would want to put it somewhere.

File: data_fixes/change_restriction/update_end_date.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      else record['notes'] = [new_accessrestrict]
+      end
+    end
+
+    post = @client.post(row['uri'], record.to_json)
+    #write to log
+    File.write(log, post.body, mode: 'a')
+    puts post.body
+    rescue Exception => msg
+    error = "Process ended: #{Time.now} with error '#{msg.class}: #{msg.message}''"
```
Comment (carolyncole) [inline data_fixes/change_restriction/update_end_date.rb:47]: Another place where the error is not being utilized

File: data_fixes/change_restriction/update_end_date.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      if record['notes'].any?
+        then record['notes'] << new_accessrestrict
+      else record['notes'] = [new_accessrestrict]
+      end
+    end
+
+    post = @client.post(row['uri'], record.to_json)
+    #write to log
+    File.write(log, post.body, mode: 'a')
+    puts post.body
```
Comment (carolyncole) [inline data_fixes/change_restriction/update_end_date.rb:45]: Did you want  to keep this puts here?

File: data_fixes/change_restriction/fix_restriction.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    #change restriction type here. Takes an array.
+    accessrestrict[0]['rights_restriction']['local_access_restriction_type'] = ["Open"]
+    #change restriction note here. Takes a string.
+    #accessrestrict[0]['subnotes'][0]['content'] = row['restriction_note']
+    #change the end date here
+    #accessrestrict[0]['rights_restriction']['end'] = row['end_date'] unless row['end_date'].empty?
+    post = @client.post(row['uri'], record.to_json)
+    #write to log
+    File.write(log, post.body, mode: 'a')
+    puts post.body
```
Comment (carolyncole) [inline data_fixes/change_restriction/fix_restriction.rb:27]: Did you want to keep this puts?

**Review by carolyncole — CHANGES_REQUESTED** [review]
@regineheberlein I'm noticing some very similar code here.  It might make you life easier in the future to extract this behavior into class.  It looks like Opening the csv file reading each record, doing manipulation of the record and them sending the record, logging the record and error handeling is the common theme.  And in each case the manipulation is the part that changes.  It's not something that needs to change with this PR of course...  If you are interested we could pair on a subsequent PR to extract the common elements.

File: data_fixes/change_restriction/update_end_date.rb (hunk 3 above)
Comment (regineheberlein) [reply data_fixes/change_restriction/update_end_date.rb:45]: yeah, that's how I keep an eye on it

File: data_fixes/change_restriction/fix_restriction.rb (hunk 4 above)
Comment (regineheberlein) [reply data_fixes/change_restriction/fix_restriction.rb:27]: yes, same as the other one

**Comment by regineheberlein** [comment]
I have no experience with object-oriented programming and would love the opportunity to pair with you on how to make this into a class!

**Pull request merged**